### DISCUSSION TITLE
On compilation things

### DISCUSSION THREAD

**Discussion by MasterInQuestion** [MasterInQuestion/talk#33] [FFmpeg]
[ marat569 @ CE 2024-06-25 01:01:34 UTC:
[ https://github.com/mpv-player/mpv/issues/14377#issuecomment-2187712265](https://github.com/mpv-player/mpv/issues/14377#issuecomment-2187712265)
    @MasterInQuestion sorry for the spam, but would you be able to build me FFmpeg with the fix in your [other post](https://github.com/mpv-player/mpv/issues/14377#issuecomment-2185315320) please: so I could try it out?
    I don't have tools setup right now to compile it myself. ]

&#x2D;---

    @marat569, I have some writing pending on compiling random programs with GCC.
    Not yet made clear for the complexity of many compilation options. (matter most of performance, compatibility)
    Moreover for the arbitrariness of the compilation scripts... [1]

    You could start by downloading GCC MinGW-w64 from:
    https://www.mingw-w64.org/downloads/#winlibscom
    (assumed Windows, for Unix-like environments typically have alike compilers bundled)
    .
    Don't use UCRT unless targeting Windows 10+. (or expect dependency hell of DLL)

    Then you may try the instructions on:
    https://trac.ffmpeg.org/wiki/CompilationGuide

    Note:
    https://github.com/FFmpeg/FFmpeg/blob/master/configure
    ; expects Unix Shell. [2]
    (so you would have to try Windows Subsystem for Linux (WSL) alike...)
[2] Most of the compilation chain, also. (e.g. ["Makefile"](https://github.com/FFmpeg/FFmpeg/blob/master/Makefile))

[ [1]
    Essentially the compilation is just alike:
    [ https://github.com/exiftool/exiftool/issues/260#issue-2254044650](https://github.com/exiftool/exiftool/issues/260#issue-2254044650)
    .
    Except it generates pre-compiled output (executable), instead of compiling on-the-fly (and run).

    Invoking the compiler properly has been wrapped of many pointless layers of sophistry, typically. ]

**Comment by marat569** [comment]
Thanks man, I'll give it a go tomorrow c:

**Reply by MasterInQuestion** [reply]
I have also linked the relevant information in mainstream FFmpeg:
    https://trac.ffmpeg.org/ticket/11072

    Fix would probably be soon.

**Reply by marat569** [reply]
Awesome! 

I need to get all this setup for future patches. I'm on a fresh windows install, and don't have stuff setup yet. Mostly just mpv and steam to play games :p
